currencies/rate_converter_test.go

```diff
@@ -27,417 +25,167 @@ func getMockRates() []byte {
 	}`)
 }
 
-func TestFetch_Success(t *testing.T) {
-
-	// Setup:
-	calledURLs := []string{}
-	mockedHttpServer := httptest.NewServer(http.HandlerFunc(
-		func(rw http.ResponseWriter, req *http.Request) {
-			calledURLs = append(calledURLs, req.RequestURI)
-			rw.WriteHeader(http.StatusOK)
-			rw.Write([]byte(getMockRates()))
-		}),
-	)
-
-	defer mockedHttpServer.Close()
+type MockClock struct {
```

Comment:
Consider naming this `FakeClock` since it doesn't have any mocking behavior and change `fakeTime` to just `time`.